applications/RomApplication/custom_strategies/custom_strategies/modal_derivative_strategy.hpp

```diff
@@ -0,0 +1,1109 @@
+                r_current_process_info[BASIS_J] = basis_j;
+
+                // Reset RHS and solution vector at each step
+                TSparseSpace::SetToZero(rb);
+                TSparseSpace::SetToZero(rDx);
+
+                // Start building RHS
+                BuiltinTimer time_build_rhs;
+                // Build RHS: -dK/dp . basis
+                p_builder_and_solver->BuildRHS(p_scheme, r_model_part, rb);
+
+                // Compute the derivative of the eigenvalue : basis^T . dK/dp . basis
+                const double deigenvalue_i_dbasis_j = -inner_prod(basis, rb);
+                r_eigenvalues[r_current_process_info[DERIVATIVE_INDEX]] = deigenvalue_i_dbasis_j;
+
+                KRATOS_INFO_IF("ModalDerivativeStrategy", (this->GetEchoLevel() >= 1 && r_model_part.GetCommunicator().MyPID() == 0)) << "Build time RHS: " << time_build_rhs.ElapsedSeconds() << std::endl;
+
+                // Apply the master-slave constraints to RHS only
+                BuiltinTimer time_rhs_constraints_and_conditions;
+                if (master_slave_constraints_defined)
+                {
+                    p_builder_and_solver->ApplyRHSConstraints(p_scheme, r_model_part, rb);
+                }
+
+                // Apply Dirichlet conditions
+                p_builder_and_solver->ApplyDirichletConditions_RHS(p_scheme, r_model_part, rDx, rb);
+                KRATOS_INFO_IF("ModalDerivativeStrategy", (this->GetEchoLevel() >= 1 && r_model_part.GetCommunicator().MyPID() == 0)) << "Apply RHS constraints and conditions: " << time_rhs_constraints_and_conditions.ElapsedSeconds() << std::endl;
+                
+                // Start Solve
+                BuiltinTimer solve;
+                // Compute particular solution
+                p_builder_and_solver->SystemSolve(rA, rDx, rb);
+
+                // Reconstruct slave DOF solution
+                if (master_slave_constraints_defined)
+                {
+                    p_builder_and_solver->ReconstructSlaveSolution(p_scheme, r_model_part, rA, rDx, rb);
+                }
+                KRATOS_INFO_IF("ModalDerivativeStrategy", (this->GetEchoLevel() >= 1 && r_model_part.GetCommunicator().MyPID() == 0)) << "System solve time: " << solve.ElapsedSeconds() << std::endl;
+
+                // Mass orthonormalization
+                if (mMassOrthonormalizeFlag)
+                {
+                    this->MassOrthonormalize(rDx);
+                }
+                
+                // Assign solution to ROM_BASIS
+                this->AssignVariables(rDx);
+
+                // Update the derivative index
+                r_current_process_info[DERIVATIVE_INDEX] += 1;
+
+                KRATOS_INFO_IF("ModalDerivativeStrategy", (this->GetEchoLevel() >= 1 && r_model_part.GetCommunicator().MyPID() == 0)) << "Compute dBasis_" << basis_i+1 << "_dalpha_" << basis_j+1 << " time: " << time_derivative.ElapsedSeconds() << std::endl;
```

Comment:
```suggestion
                KRATOS_INFO_IF("ModalDerivativeStrategy", this->GetEchoLevel() >= 1) << "Compute dBasis_" << basis_i+1 << "_dalpha_" << basis_j+1 << " time: " << time_derivative.ElapsedSeconds() << std::endl;
```

the Logger can handle this internally now

Reply:
so you suggest the same change for each call?

Reply:
that would be the cleanes